protocols/imap/src/main/java/org/apache/james/imap/message/request/ListRequest.java

```diff
@@ -46,11 +68,25 @@ public final String getMailboxPattern() {
         return mailboxPattern;
     }
 
+    public EnumSet<ListSelectOption> getSelectOptions() {
+        return selectOptions;
+    }
+
+    public EnumSet<ListReturnOption> getReturnOptions() {
+        return returnOptions;
+    }
+
+    public final boolean selectSubscribed() {
+        return getSelectOptions().contains(ListSelectOption.SUBSCRIBED);
+    }
+
     @Override
     public String toString() {
         return MoreObjects.toStringHelper(this)
             .add("baseReferenceName", baseReferenceName)
             .add("mailboxPattern", mailboxPattern)
+            .add("selectOptions", selectOptions.stream().map(Enum::toString).collect(Collectors.joining(",")))
+            .add("returnOptions", returnOptions.stream().map(Enum::toString).collect(Collectors.joining(",")))
```

Comment:
```suggestion
            .add("selectOptions", selectOptions)
            .add("returnOptions", returnOptions)
```

EnumSet default toString is displaying good enough and is more efficient.